packages/components/cascader-panel/src/index.vue

```diff
@@ -150,21 +153,37 @@ const expandNode: ElCascaderPanelContext['expandNode'] = (node, silent) => {
     !silent && emit('expand-change', node?.pathValues || [])
   }
 }
-
+const handleBeforeChange = props.beforeChange ?? (() => true)
```

Comment:
Put this variable inside the `handleCheckChange` function so that even if `props.beforeChange` is modified dynamically, the latest value can be obtained.